### ISSUE TITLE
Error: spawn cmd.exe ENOENT on every keypress

### ISSUE THREAD

**Issue by PawelWozniak** [AtomLinter/linter-flake8#209]
Hi,
I have installed Atom 1.8.0 on Windows 7 64bit by using AtomSetup.exe .
Installed also Node.js v4.4.5 and flake8 by pip install flake8 - install output:
Successfully installed flake8-2.6.0 mccabe-0.5.0 pycodestyle-2.0.0 pyflakes-1.2.3

Then installed linter and linter-flake8 on atom.
When opening python file on every keypress I am getting red box with error:

```
Error: spawn cmd.exe ENOENT
    at exports._errnoException (util.js:856:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at doNTCallback2 (node.js:465:9)
    at process._tickCallback (node.js:379:17)
```

Done already search in google but nothing usefull found.
cmd.exe exists in C:\Windows\System32\cmd.exe

Please help solving that.

**Comment by mostaphaRoudsari** [comment]
This also happens to me after updating to the latest version on Windows 8. The previous version was working fine.

**Comment by Arcanemagus** [comment]
This is most likely due to a bug in `atom-linter@^5`, will try to trace that down...

**Comment by mostaphaRoudsari** [comment]
I don't see this error anymore. Works as expected!

**Comment by MurKit** [comment]
I have the same problem

**Comment by thiagoiglesias** [comment]
I'm having the same issue, but I'm running it on a Mac. Any ideas on how to fix it?

**Comment by stueken** [comment]
Same problem, but running it on Linux.

**Comment by ssbarnea** [comment]
I do get exactly the same error on Atom 1.8 on MacOS making the entire plugin useless.
<img width="469" alt="screenshot 2016-07-23 12 33 29" src="https://cloud.githubusercontent.com/assets/102495/17077718/49928bd8-50d2-11e6-8668-5b86100e09e2.png">

**Comment by Arcanemagus** [comment]
@ssbarnea and @thiagoiglesias as this issue is Windows specific, it's pretty impossible for you to be hitting the _exact_ same issue 😛.

@ssbarnea is `/usr/local/bin/flake8` what you get when you run `which flake8`?

@thiagoiglesias Try specifying the _full_ path to `flake8` in your settings.

**Comment by thiagoiglesias** [comment]
@Arcanemagus, thanks for the answer. Sorry for my ignorance, but how can I figure ou the full path to flake8 to specify?

**Comment by ssbarnea** [comment]
And the generic way to find it is `which flake8`, but again I would recomment calling it using python `python3 -m flake8` (you can use which on python too).

For example on my machine I would use `/usr/local/bin/python3 -m flake8`

**Comment by Arcanemagus** [comment]
@ssbarnea Because of the way `spawn()` works, you can't enter that as the executable path though.

**Comment by ssbarnea** [comment]
@Arcanemagus That's why you need to split the command line before you feed it to the spawn(). Is is known that on Unix in order to start a process you need to send parameters as an array instead of the "windows" kind of call which does support a simple command.

JavaScript is not among my skillset but I gues you could use something like https://www.npmjs.com/package/parse-spawn-args to parse the command line.

**Comment by Arcanemagus** [comment]
`atom-linter@8`, included in v1.13.3 fixes this issue on Windows.

@ssbarnea if you are still seeing this with the latest version can you please file a _new_ issue?

**Comment by sudhanvadp** [comment]
Just install yarn before react-native init command. It is as simple as that!!!
